File: DeviantArtFs/Types/ResponseTypes/DeviationFile.fs
namespace DeviantArtFs

open FSharp.Json

type IDeviationFile =
    abstract member Src: string
    abstract member Height: int
    abstract member Width: int

type DeviationPreview = {
    src: string
    height: int
    width: int
    transparency: bool
} with
    interface IDeviationFile with
        member this.Src = this.src
        member this.Height = this.height
        member this.Width = this.width

type DeviationDownload = {
    src: string
    height: int
    width: int
    filesize: int
} with
    static member Parse json = Json.deserialize<DeviationDownload> json
    interface IDeviationFile with
        member this.Src = this.src
        member this.Height = this.height
        member this.Width = this.width

type DeviationContent = {
    src: string
    height: int
    width: int
    transparency: bool
    filesize: int
} with
    interface IDeviationFile with
        member this.Src = this.src
        member this.Height = this.height
        member this.Width = this.width

File: DeviantArtFs.Types/DeviationFile.cs
using System;

namespace DeviantArtFs.Types {
    public abstract class DeviationFile {
        public string src { get; set; } = "";
        public int width { get; set; }
        public int height { get; set; }
    }

    public class DeviationPreview : DeviationFile {
        public int filesize { get; set; }
    }

    public class DeviationContent : DeviationPreview {
        public bool transparency { get; set; }
    }

    public class DeviationFlash : DeviationFile { }
}
